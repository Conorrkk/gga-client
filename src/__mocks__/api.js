const teamMock = {
    teamName: 'Bredagh',
    teamLevel: 'Senior',
}

export const apiRequests = () => {
    return {
        getTotalGoals: jest.fn(() => 2),
        getTotalPoints: jest.fn(() => 10),
        getTeamById: jest.fn(() => {
            return teamMock;
        })
    };
}