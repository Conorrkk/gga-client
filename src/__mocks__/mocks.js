export const player1 = {
    _id: 1,
    playerId: 1,
    playerName: 'Conor Kelly',
    stats: {
        goal_from_play: 0,
        goal_from_dead: 1,
        point_from_play: 2,
        point_from_dead: 0,
        wide: 0
    }
}

export const player2 = {
    _id: 2,
    playerId: 2,
    playerName: 'Harry Potter',
    stats: {
        goal_from_play: 0,
        goal_from_dead: 1,
        point_from_play: 2,
        point_from_dead: 0,
        wide: 2
    }
}

export const playersMock = [player1, player2]; 

export const match = {
    _id: "123",
    teams: {
        teamId: "123",
        oppositionTeam: "Galway",
        players: [player1, player2]
    },
    matchDate: 8/30/2023,
    goalAgainst: 1,
    pointAgainst: 14
};

export const matches = [match];