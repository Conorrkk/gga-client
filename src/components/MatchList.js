import MatchShow from "./MatchShow";
import { Container } from "react-bootstrap";

function MatchList({ matches, onDelete }) {
  const loadedMatches = matches.map((match) => {
    return <MatchShow onDelete={onDelete} key={match._id} match={match} />;
  });

  return <Container fluid >{loadedMatches}</Container>;
}

export default MatchList;
