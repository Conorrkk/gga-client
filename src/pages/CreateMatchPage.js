import React, { useState, useEffect, useContext } from "react";
import { getTeams, postMatch } from "../api";
import { Form, Button } from "react-bootstrap";
import { useNavigate } from "react-router-dom";
import CurrentMatchContext from "../context/CurrentMatchProvider";
import NavBar from "../components/NavBar";
// import Footer from "../components/Footer";

function CreateMatchPage() {
  // array of users teams
  const [usersTeams, setUsersTeams] = useState([]);
  // current team
  const [team, setTeam] = useState("");
  // string of opponent name to go in db
  const [opponent, setOpponent] = useState("");
  // for setting the context to the match being created
  const [currentMatch, setCurrentMatch] = useContext(CurrentMatchContext);

  // for nav
  const navigate = useNavigate();

  // gets the users teams and sets them as state
  useEffect(() => {
    getTeams()
      .then((response) => {
        // if there is a response and its an array then we set the usersTeams array
        if (response && Array.isArray(response.teams)) {
          setUsersTeams(response.teams);
        } else {
          console.log("Issue with getTeams() response", response);
        }
      })
      .catch((error) => {
        console.error("Error getting users teams:", error);
      });
  }, []);

  // when user selects a team set this team as the current team state
  const handleTeamChange = (event) => {
    const selTeamId = event.target.value;
    const teamToSelect = usersTeams.find((team) => team._id === selTeamId);
    setTeam(teamToSelect);
  };

  // when user changes the opposing teams name update the opponent state to this new value
  const handleChangeOpponent = (event) => {
    setOpponent(event.target.value);
  };

  // when user clicks select panel, create match in db & navigate to choose panel page
  const handleSubmit = (event) => {
    event.preventDefault();
    const match = {
      team,
      opponent,
    };
    postMatch(match)
      .then((response) => {
        // for setting current match so that stats can be recorded
        setCurrentMatch(response);
        navigate(`/match/${response._id}/team/${team._id}`);
      })
      .catch((error) => {
        console.error("Error creating match:", error);
      });
  };

  return (
    <div>
      <NavBar />
      <div className="login d-flex justify-content-center align-items-center vh-100 bg-white">
        <div className="form_container p-5 rounded bg-white border border-dark border-2">
          <Form onSubmit={handleSubmit}>
            <div className="mx-2 my-2">
              <Form.Group className="mb-2" controlId="team">
                <Form.Label>Choose your team</Form.Label>
                <Form.Select
                  value={team ? team._id : ""}
                  onChange={handleTeamChange}
                >
                  <option value="">Select team</option>
                  {usersTeams.map((team) => (
                    <option key={team._id} value={team._id}>
                      {team.teamName} {team.teamLevel} Team
                    </option>
                  ))}
                </Form.Select>
              </Form.Group>
            </div>
            <div className="mx-2 my-2">
              <Form.Group className="mb-2" controlId="opponentName">
                <Form.Label>Opposing team</Form.Label>
                <Form.Control
                  value={opponent}
                  onChange={handleChangeOpponent}
                  type="text"
                  autoComplete="off"
                  placeholder="Team Name"
                />
              </Form.Group>
            </div>
            <div className="mx-2 my-2">
              {/* <Link to={`/chooseMatchPanel/${team._id}`}>
                <Button variant="outline-success" type="submit">
                  Select panel
                </Button>
              </Link> */}
              <Button variant="outline-success" type="submit">
                Select panel
              </Button>
            </div>
          </Form>
        </div>
      </div>
      {/* <Footer /> */}
    </div>
  );
}

export default CreateMatchPage;
