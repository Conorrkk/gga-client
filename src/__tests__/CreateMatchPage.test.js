import React from 'react';
import renderer from 'react-test-renderer';
import MatchList from '../components/MatchList';
import { matches } from '../__mocks__/mocks';
import { BrowserRouter } from 'react-router-dom';
import CreateMatchPage from '../pages/CreateMatchPage';
import CurrentMatchContext, { MatchProvider } from '../context/CurrentMatchProvider';
import { render } from 'react-dom';


it('renders correctly', () => {

    const tree = renderer.create(
        <BrowserRouter>
            <MatchProvider>
                <CreateMatchPage />
            </MatchProvider>
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});