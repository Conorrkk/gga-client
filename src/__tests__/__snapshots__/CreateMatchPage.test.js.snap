// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <nav
      className="custom-navbar navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <a
        className="ms-2 navbar-brand"
        href="#"
      >
        GGAnalytics
      </a>
      <button
        aria-controls="navbarScroll"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="navbarScroll"
      >
        <div
          className="mr-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
          onKeyDown={[Function]}
          style={
            Object {
              "maxHeight": "100px",
            }
          }
        >
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            disabled={false}
            href="/dashboard"
            onClick={[Function]}
          >
            Dashboard
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            disabled={false}
            href="/teams"
            onClick={[Function]}
          >
            Teams
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            disabled={false}
            href="/createMatch"
            onClick={[Function]}
          >
            Create Match
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            disabled={false}
            href="/matchHistory"
            onClick={[Function]}
          >
            Match History
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Logout
          </a>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="container-fluid"
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-6 offset-xl-3 offset-lg-3 offset-md-3 offset-sm-3"
    >
      <div
        className="styled-card card"
        style={
          Object {
            "marginTop": "10vh",
          }
        }
      >
        <h5
          className="card-header"
        >
          New Match
        </h5>
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            <form
              className=""
              onSubmit={[Function]}
            >
              <div
                className="mx-2 my-2"
              >
                <div
                  className="mb-2"
                >
                  <label
                    className="form-label"
                    htmlFor="team"
                  >
                    Select match team
                  </label>
                  <select
                    className="form-select"
                    id="team"
                    onChange={[Function]}
                    required={true}
                    value=""
                  >
                    <option
                      value=""
                    >
                      Select team
                    </option>
                  </select>
                </div>
              </div>
              <div
                className="mx-2 my-2"
              >
                <div
                  className="mb-2"
                >
                  <label
                    className="form-label"
                    htmlFor="opponentName"
                  >
                    Opposing team
                  </label>
                  <input
                    autoComplete="off"
                    className="form-control"
                    id="opponentName"
                    maxLength="60"
                    onChange={[Function]}
                    pattern="[a-zA-Z0-9\\\\s]+"
                    placeholder="Team Name"
                    required={true}
                    title="Only letters and numbers(max 60 chars)"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="mx-2 my-2"
              >
                <button
                  className="styled-button btn btn-outline-primary"
                  disabled={false}
                  type="submit"
                >
                  Create
                  <svg
                    className="styled-icon-button"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
                    />
                  </svg>
                </button>
              </div>
            </form>
          </p>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="main-footer bg-dark text-white"
  >
    <div
      className="mt-2 container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <p>
            Â© 
            2023
             GGAnalytics
          </p>
        </div>
        <div
          className="col"
        >
          <div
            className="vstack gap-2"
          >
            <h3>
              Quick Links
            </h3>
            <a
              href="/dashboard"
              onClick={[Function]}
            >
              <div
                className="p-2"
              >
                Dashboard
              </div>
            </a>
            <a
              href="/teams"
              onClick={[Function]}
            >
              <div
                className="p-2"
              >
                Teams
              </div>
            </a>
            <a
              href="/matchHistory"
              onClick={[Function]}
            >
              <div
                className="p-2"
              >
                Match History
              </div>
            </a>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="vstack gap-2"
          >
            <h3>
              Contact
            </h3>
            <div
              className="p-2"
            >
              +353863032593
            </div>
            <div
              className="p-2"
            >
              gganalytics@gga.project
            </div>
            <div
              className="p-2"
            >
              Computer Science Building, Belfast, United Kingdom, BT9 5AF
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
